api/tests/Feature/UserTest.php

```diff
@@ -2154,4 +2154,223 @@ public function testFilterByClassificationToSalaryWithPools(): void
+        $this->graphQL(/** @lang Graphql */ '
+            query getUsersPaginated($where: UserFilterAndOrderInput) {
+                usersPaginated(where: $where) {
+                    paginatorInfo {
+                        total
+                    }
+                }
+            }
+        ', [
+            'where' => [
+                'name' => 'r',
+            ]
+        ])->assertJson([
+            'data' => [
+                'usersPaginated' => [
+                    'paginatorInfo' => [
+                        'total' => 3
+                    ]
+                ]
+            ]
+        ]);
+
+        // ensure queries with multiple filter variables apply separately as AND operations (builds off assertion above)
+        $this->graphQL(/** @lang Graphql */ '
+            query getUsersPaginated($where: UserFilterAndOrderInput) {
+                usersPaginated(where: $where) {
+                    paginatorInfo {
+                        total
+                    }
+                }
+            }
+        ', [
+            'where' => [
+                'name' => 'r',
+                'telephone' => "12345",
+            ]
+        ])->assertJson([
+            'data' => [
+                'usersPaginated' => [
+                    'paginatorInfo' => [
+                        'total' => 1
+                    ]
+                ]
+            ]
+        ]);
+
+        // Assert email filter with partial email returns correct count
+        $this->graphQL(/** @lang Graphql */ '
+            query getUsersPaginated($where: UserFilterAndOrderInput) {
+                usersPaginated(where: $where) {
+                    paginatorInfo {
+                        total
+                    }
+                }
+            }
+        ', [
+            'where' => [
+                'email' => 'user.com',
+            ]
+        ])->assertJson([
+            'data' => [
+                'usersPaginated' => [
+                    'paginatorInfo' => [
+                        'total' => 4
+                    ]
+                ]
+            ]
+        ]);
+
+        // Assert filtering for phone digit in general search returns correct count
+        $this->graphQL(/** @lang Graphql */ '
+            query getUsersPaginated($where: UserFilterAndOrderInput) {
+                usersPaginated(where: $where) {
+                    paginatorInfo {
+                        total
+                    }
+                }
+            }
+        ', [
+            'where' => [
+                'generalSearch' => '9',
+            ]
+        ])->assertJson([
+            'data' => [
+                'usersPaginated' => [
+                    'paginatorInfo' => [
+                        'total' => 2
+                    ]
+                ]
+            ]
+        ]);
+
+        // Assert filtering for last name in general search returns correct count
+        $this->graphQL(/** @lang Graphql */ '
```

Comment:
I'm most concerned about how the name query works, so could we have a few more tests for that?
eg
test that a full name works (bob rob returns 1 result)
test that two name segments work (bo ro returns 1 result)
test that name segments work in incorrect order (ro bo returns 1 result)

Reply:
added